src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1443,7 +1443,7 @@ private void WritePackageDependenciesDesignTime()
                 foreach (var package in _lockFile.Libraries)
                 {
                     var packageVersion = package.Version.ToNormalizedString();
-                    string packageId = $"{package.Name}/{packageVersion}";
+                    string packageId = $"{package.Name}/{packageVersion.ToLowerInvariant()}";
```

Comment:
@nkolev92 Does the lockFile.Libraries folder hold the same assumption as the global packages folder which is that all of the versions will be lowercased? 

This applies to the ResolvePackageDependencies file as well.

Reply:
I don't think the libraries lowercases them.
It uses the true casing from nuspec afaik. 
The expectation is that the comparison is case insensitive there. 

I actually think that within the assets file, the id of the package will always have the consistent case.

Reply:
Is it ok then if I don't lowercase them then?

Reply:
If you check the usage of packageId, it's used in allowItemSpecs.Contains which is an OrdinalIgnoreCase hashset.

Reply:
It should be ok but I would leave it as is as Nikolche suggests ; I think putting this conversation in a comment would be helpful.